modules/javafx.graphics/src/main/java/javafx/scene/layout/Region.java

```diff
@@ -2058,62 +2028,133 @@ public final double snappedRightInset() {
         }
     }
 
+    /*
+     * Definition of used terms:
+     *
+     * # available width/heights:
+     *
+     * Sizes provided by the container that may be used as a dependent value when
+     * calculating sizes for biased controls. These may be set to -1 to indicate
+     * no such information is available. If given, the sizes include the Margin
+     * of the child. As such the Margin must be removed before passing these
+     * values as a dependent value to min/pref/max width/height functions.
+     *
+     * # content width/heights:
+     *
+     * The space allocated to a child, minus its margins. These are never -1.
+     *
+     * # bounded width/heights:
+     *
+     * The space allocated to a child, minus its margins, adjusted according to
+     * its constraints (min <= X <= max). These are never -1.
```

Comment:
always `> 0.0` here?

Reply:
0 is probably still allowed, but it shouldn't be negative.